generators/java/sdk/src/main/java/com/fern/java/client/generators/endpoint/HttpUrlBuilder.java

```diff
@@ -168,23 +187,37 @@ private boolean addHttpPathToCodeBlock(
                         PoetTypeNameStringifier.stringify(
                                 clientOptionsField.name + "." + variableGetter.name + "()",
                                 poetPathParameter.poetParam().type));
-            } else if (typeNameIsOptional(poetPathParameter.poetParam().type)) {
-                codeBlock.add(";\n");
-                codeBlock
-                        .beginControlFlow("if ($L.isPresent())", poetPathParameter.poetParam().name)
-                        .addStatement(
-                                "$L.addPathSegment($L)",
-                                httpUrlname,
-                                PoetTypeNameStringifier.stringify(
-                                        poetPathParameter.poetParam().name + ".get()",
-                                        poetPathParameter.poetParam().type))
-                        .endControlFlow();
-                endedWithStatement = true;
             } else {
-                codeBlock.add(
-                        "\n.addPathSegment($L)",
-                        PoetTypeNameStringifier.stringify(
-                                poetPathParameter.poetParam().name, poetPathParameter.poetParam().type));
+                String paramName = poetPathParameter.poetParam().name;
+                if (inlinePathParams
+                        && poetPathParameter.irParam().getLocation().equals(PathParameterLocation.ENDPOINT)
+                        && httpEndpoint.getSdkRequest().isPresent()) {
+                    paramName = httpEndpoint
+                                    .getSdkRequest()
+                                    .get()
+                                    .getRequestParameterName()
+                                    .getCamelCase()
+                                    .getSafeName()
+                            // TODO(agateno): How do we get the getter name from the request body file?
```

Comment:
Do we need to resolve this here, or should be an afterthought?

Reply:
I don't think it's make or break given that we generate all the getters in a predictable format, but if we could code-share that here it'd be cleaner.